src/main/java/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package onlinebookstore.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import onlinebookstore.dto.order.CreateOrderRequestDto;
+import onlinebookstore.dto.order.CreateOrderResponseDto;
+import onlinebookstore.dto.order.OrderItemResponseDto;
+import onlinebookstore.dto.order.OrderResponseDto;
+import onlinebookstore.dto.order.OrderStatusResponseDto;
+import onlinebookstore.dto.order.UpdateStatusRequestDto;
+import onlinebookstore.exception.EntityNotFoundException;
+import onlinebookstore.mapper.OrderItemMapper;
+import onlinebookstore.mapper.OrderMapper;
+import onlinebookstore.model.CartItem;
+import onlinebookstore.model.Order;
+import onlinebookstore.model.OrderItem;
+import onlinebookstore.model.ShoppingCart;
+import onlinebookstore.model.Status;
+import onlinebookstore.model.User;
+import onlinebookstore.repository.OrderItemRepository;
+import onlinebookstore.repository.OrderRepository;
+import onlinebookstore.repository.ShoppingCartRepository;
+import onlinebookstore.repository.UserRepository;
+import onlinebookstore.service.OrderService;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public CreateOrderResponseDto createOrder(CreateOrderRequestDto orderRequestDto) {
+        Order order = initializeOrder(orderRequestDto,
+                shoppingCartRepository.findByUserId(getUser().getId()).orElseThrow(() ->
```

Comment:
It's better to create variable for this. 
Bad practice to make logic instead of method argument